internal/agent/device/spec/spec_test.go

```diff
@@ -93,6 +95,126 @@ func TestBootstrapCheckRollback(t *testing.T) {
 
 }
 
+func TestNewManager(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockReadWriter := fileio.NewMockReadWriter(ctrl)
+	mockBootcClient := container.NewMockBootcClient(ctrl)
+	logger := log.NewPrefixLogger("test")
+	backoff := wait.Backoff{}
+
+	t.Run("constructs file paths for the spec files", func(t *testing.T) {
+		manager := NewManager(
+			"device-name",
+			"test/directory/structure/",
+			mockReadWriter,
+			mockBootcClient,
+			backoff,
+			logger,
+		)
+
+		require.Equal("test/directory/structure/current.json", manager.currentPath)
+		require.Equal("test/directory/structure/desired.json", manager.desiredPath)
+		require.Equal("test/directory/structure/rollback.json", manager.rollbackPath)
+	})
+}
+
+func TestInitialize(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockReadWriter := fileio.NewMockReadWriter(ctrl)
+
+	s := &SpecManager{
+		deviceReadWriter: mockReadWriter,
+	}
+
+	writeErr := errors.New("write failure")
+
+	t.Run("error writing current file", func(t *testing.T) {
+		// current
+		mockReadWriter.EXPECT().WriteFile(gomock.Any(), gomock.Any(), gomock.Any()).Return(writeErr)
+		err := s.Initialize()
+		require.ErrorContains(err, "writing current rendered spec:")
```

Comment:
This can be a followup but we should always strive to test for error types. So if we see a common error lets create a an error like

```go
var ErrWritingSpec = errors.New("writing current spec") 
```
Then we can wrap `%w` that error on the return and explicitly check for the type. This is much less brittle for the cases where the error message changes.

```go
return fmt.Errorf("%w: current: %w", ErrWritingSpec, err)
```
Then testing is 

```go
require.ErrorIs(err, ErrWritingSpec)
```